File: practice/algorithms/sorting/quicksort_1_partition/quicksort_1_partition.hs
main = do
    n <- readLn :: IO Int
    ar <- getLine
    let a = map (read :: String -> Int) $ words ar
        p = a !! 0
        smaller = filter  (<p) a
        equal   = filter (==p) a
        greater = filter  (>p) a
        total = smaller ++ equal ++ greater
    putStrLn $ unwords $ map show total

File: practice/algorithms/sorting/quicksort_1_partition/quicksort_1_partition.py
#!/usr/bin/env python

def partition(ar):
    p = ar[0]
    smaller = filter(lambda x : x < p, ar)
    equal   = filter(lambda x : x == p, ar)
    greater = filter(lambda x : x > p, ar)
    total = smaller + equal + greater
    for VAL in total:
        print VAL,
    return ""

m = input()
ar = [int(i) for i in raw_input().strip().split()]
partition(ar)

